tests/integration/utils.py

```diff
@@ -178,3 +217,41 @@ def __check_attributes_schema(actual_attributes, expected_attributes):
                 __check_error_schema(error)
     except KeyError as error:
         self.fail(error)
+
+
+# Check response of an endpoint against path cases for schema, data validity
+def test_path_request(self, endpoint, resource, response_code, test_cases,
+                      param=None, test_assertion=None):
+    schema = get_resource_schema(self, resource)
+    for test_case in test_cases:
+        response = make_request(self, f'{endpoint}/{test_case}',
+                                response_code)
+        check_schema(self, response, schema)
+        if test_assertion:
+            actual_case = response.json()['data']['attributes'][param]
+            test_assertion(self, actual_case, test_case)
+
+
+# Check response of an endpoint against query cases for schema, data validity
+def test_query_request(self, endpoint, resource, response_code, test_cases,
+                       param, test_assertion=None):
+    schema = get_resource_schema(self, resource)
+    for test_case in test_cases:
+        response = make_request(self, endpoint, response_code,
+                                params={param: test_case})
+        check_schema(self, response, schema)
+        if test_assertion:
+            for resource in response.json()['data']:
+                actual_case = resource['attributes'][param]
+                test_assertion(self, actual_case, test_case)
+
```

Comment:
As we discussed, these two functions should be improved by combining the similar logic together. Also, some variables naming are confusing (e.g. `param`), can you also make them more clear?